great_expectations/experimental/datasources/data_asset/data_connector/filesystem_data_connector.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+import re
+from typing import List
+
+from great_expectations.datasource.data_connector.util import (
+    get_filesystem_one_level_directory_glob_path_list,
+    normalize_directory_path,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.file_path_data_connector import (
+    FilePathDataConnector,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class FilesystemDataConnector(FilePathDataConnector):
+    """Extension of ConfiguredAssetFilePathDataConnector used to connect to Filesystem.
+
+    Being a Configured Asset Data Connector, it requires an explicit list of each Data Asset it can
+    connect to. While this allows for fine-grained control over which Data Assets may be accessed,
+    it requires more setup.
+
+    Args:
+        name (str): name of ConfiguredAssetFilesystemDataConnector
+        regex (pattern): Optional regex pattern for filtering data references
+        glob_directive (str): glob for selecting files in directory (defaults to `**/*`) or nested directories (e.g. `*/*/*.csv`)
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+        # TODO: <Alex>ALEX</Alex>
+        # sorters (list): Optional list if you want to sort the data_references
+        # TODO: <Alex>ALEX</Alex>
+    """
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        data_asset_name: str,
+        base_directory: pathlib.Path,
+        regex: re.Pattern,
+        glob_directive: str = "**/*",
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+        # TODO: <Alex>ALEX</Alex>
+        # sorters: Optional[list] = None,
+        # TODO: <Alex>ALEX</Alex>
+    ) -> None:
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            data_asset_name=data_asset_name,
+            regex=regex,
+            # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+            # TODO: <Alex>ALEX</Alex>
+            # sorters=sorters,
+            # TODO: <Alex>ALEX</Alex>
+            file_path_template_map_fn=None,
+        )
+
+        self._base_directory = base_directory
+        self._glob_directive: str = glob_directive
+
+    @property
+    def base_directory(self) -> pathlib.Path:
+        """
+        Accessor method for base_directory. If directory is a relative path, interpret it as relative to the
+        root directory. If it is absolute, then keep as-is.
+        """
+        return normalize_directory_path(
```

Comment:
can this be defined in a path-aware base class? It's repeated from the other connector

Reply:
@donaldheppner I am not sure exactly what is being suggested here; could you please elaborate and/or provide an example or a reference?  Please note that there is no repetition; we are calling a utility class in V3, made capable to serve both V3 and ZEP usage patterns; that method provides some checks, hence encapsulating made sense.  Thanks.

Reply:
We have two file-based data connectors. `filesystem_data_connector` and `file_path_data_connector`, and logic is duplicated between them. A common, abstract `file_data_connector` could encapsulate this shared logic

Reply:
@donaldheppner No logic should be duplicated -- could you please point to what you are seeing?  Formally,   `FilesystemDataConnector` is an implementation for the filesystem case of the general `FilePathDataConnector`, which is also the parent class of `S3DataConnector` -- `FilePathDataConnector` defines (or propagates down) the interfaces.  Happy to discuss and pair on the code review to insure we are missing anything.  Thanks.